analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -1637,3 +1637,32 @@ public void BoolSymbols_FalseFalse()
         public void FailsWhenFalseAny([DoesNotReturnIf(false)] bool condition1, [DoesNotReturnIf(false)] bool condition2) { }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/6176
+public class Repro_6176
+{
+    public void ExpressionArguments(object arg)
+    {
+        Select(x => x.FirstOrDefault().ToString());         // Compliant - custom expression processor, like Entity Framework, can propagate the "null" without executing the actual code
+        Select(() => arg == null ? arg.ToString() : null);  // Compliant, we don't analyze arguments of expression
+
+        System.Linq.Expressions.Expression<Func<IEnumerable<object>, object>> expr;
+        expr = x => x.FirstOrDefault().ToString();          // Compliant as well
+    }
+
+    public void DelegateArguments(object arg)
+    {
+        Invoke(x => x.FirstOrDefault().ToString());         // Noncompliant
+        Invoke(() => arg == null ? arg.ToString() : null);  // Noncompliant
+    }
+
+    public void Nested(object arg)
+    {
+        Select(() => Invoke(x => x.FirstOrDefault().ToString()));   // Compliant, it's invoked lambda inside an expression tree
+    }
+
+    private void Select(System.Linq.Expressions.Expression<Func<IEnumerable<object>, object>> expression) { }
+    private void Select(System.Linq.Expressions.Expression<Func<object>> expression) { }
+    private object Invoke(Func<IEnumerable<object>, object> expression) => null;
+    private void Invoke(Func<object> expression) { }
```

Comment:
It would be better if the names reflect whether they operate on `Func` or `Expression<Func>`:
```suggestion
    private void SelectExpression(System.Linq.Expressions.Expression<Func<IEnumerable<object>, object>> expression) { }
    private void SelectExpression(System.Linq.Expressions.Expression<Func<object>> expression) { }
    private object Select(Func<IEnumerable<object>, object> expression) => null;
    private void Select(Func<object> expression) { }
```